server/gokbg3/grails-app/services/wekb/KbartImportService.groovy

```diff
@@ -0,0 +1,1148 @@
+
+            if(tippList.size() == 1){
+                log.debug("tippMatchingByTitleID provider internal identifier matching by "+tippList.size() + ": "+ tippList.id)
+                return tippList[0]
+            }
+        }
+
+    }
+
+    /**
+     * Create a new TIPP
+     */
+    TitleInstancePackagePlatform tippCreate(tipp_fields = [:]) {
+
+        def tipp_status = tipp_fields.status ? RefdataCategory.lookup(RCConstants.KBCOMPONENT_STATUS, tipp_fields.status) : null
+
+        RefdataValue tipp_medium = null
+        if (tipp_fields.medium) {
+            tipp_medium = determineMediumRef(tipp_fields)
+        }
+        RefdataValue tipp_publicationType = null
+        if (tipp_fields.type) {
+            tipp_publicationType = determinePublicationType(tipp_fields)
+        }
+
+        TitleInstancePackagePlatform result = new TitleInstancePackagePlatform(
+                uuid: tipp_fields.uuid,
+                status: tipp_status,
+                name: tipp_fields.name,
+                medium: tipp_medium,
+                publicationType: tipp_publicationType,
+                url: tipp_fields.url).save(failOnError: true)
+        if (result) {
+
+            def pkg_combo_type = RefdataCategory.lookupOrCreate(RCConstants.COMBO_TYPE, 'Package.Tipps')
+            new Combo(toComponent: result, fromComponent: tipp_fields.pkg, type: pkg_combo_type).save(flush: true, failOnError: true)
+
+            def plt_combo_type = RefdataCategory.lookupOrCreate(RCConstants.COMBO_TYPE, 'Platform.HostedTipps')
+            new Combo(toComponent: result, fromComponent: tipp_fields.hostPlatform, type: plt_combo_type).save(flush: true, failOnError: true)
+
+        } else {
+            log.error("TIPP creation failed!")
+        }
+
+        result
+    }
+
+    static RefdataValue determinePublicationType(tippObj) {
+        if (tippObj.type) {
+            switch (tippObj.type) {
+                case "serial":
+                case "Serial":
+                case "Journal":
+                case "journal":
+                    return RDStore.TIPP_PUBLIC_TYPE_SERIAL
+                    break;
+                case "monograph":
+                case "Monograph":
+                case "Book":
+                case "book":
+                    return RDStore.TIPP_PUBLIC_TYPE_MONO
+                    break;
+                case "Database":
+                case "database":
+                    return RDStore.TIPP_PUBLIC_TYPE_DB
+                    break;
+                case "Other":
+                case "other":
+                    return RDStore.TIPP_PUBLIC_TYPE_OTHER
+                    break;
+                default:
+                    return null
+                    break;
+            }
+        }
+        else {
+            return null
+        }
+    }
+
+    static RefdataValue determineMediumRef(titleObj) {
```

Comment:
Did we consider LAS:eR when extending the refdata category? How much sense does it make at all to restrict it to a controlled list?